addons/account/models/account_tax.py

```diff
@@ -1382,6 +1489,22 @@ class AccountTaxRepartitionLine(models.Model):
 
     tag_ids_domain = fields.Binary(string="tag domain", help="Dynamic domain used for the tag that can be set on tax", compute="_compute_tag_ids_domain")
 
+    @api.model_create_multi
+    def create(self, vals):
+        tax_ids = list(set([line.get('tax_id') for line in vals]))
+        taxes = self.env['account.tax'].search([('id', 'in', tax_ids)])
+        tax_used = dict(taxes.mapped(lambda tax: (tax.id, tax.is_used)))
+        for line in vals:
+            if tax_used.get(line.get('tax_id'), False):
+                raise ValidationError(_("This tax has already been used, you cannot add nor delete its tax repartition lines."))
```

Comment:
"This tax", which one? You are in a for loop. Also a good idea to say used where if one needs to take action.